contracts/modules/delegates/IexecOrderManagementDelegate.sol

```diff
@@ -119,4 +121,43 @@ contract IexecOrderManagementDelegate is IexecOrderManagement, DelegateBase, Sig
             emit ClosedRequestOrder(requestorderHash);
         }
     }
+
+    function computeDealVolume(
+        IexecLibOrders_v5.AppOrder calldata appOrder,
+        IexecLibOrders_v5.DatasetOrder calldata datasetOrder,
+        IexecLibOrders_v5.WorkerpoolOrder calldata workerpoolOrder,
+        IexecLibOrders_v5.RequestOrder calldata requestOrder
+    ) external view override returns (uint256 volume) {
+        return
+            _computeVolume(
+                appOrder.volume,
+                _toTypedDataHash(appOrder.hash()),
+                datasetOrder.dataset != address(0),
+                datasetOrder.volume,
+                _toTypedDataHash(datasetOrder.hash()),
+                workerpoolOrder.volume,
+                _toTypedDataHash(workerpoolOrder.hash()),
+                requestOrder.volume,
+                _toTypedDataHash(requestOrder.hash())
+            );
+    }
+
+    function _computeVolume(
+        uint256 apporderVolume,
+        bytes32 appOrderTypedDataHash,
+        bool hasDataset,
+        uint256 datasetorderVolume,
+        bytes32 datasetOrderTypedDataHash,
+        uint256 workerpoolorderVolume,
+        bytes32 workerpoolOrderTypedDataHash,
+        uint256 requestorderVolume,
+        bytes32 requestOrderTypedDataHash
+    ) internal view returns (uint256 volume) {
+        volume = apporderVolume - m_consumed[appOrderTypedDataHash];
```

Comment:
Do we stack too deep if returning directly (instead of using tmp var)?

Reply:
Just FYI, we will for sure stack too deep with my coming PR.

Reply:
Do you mean this way ? 
`volume = Math.min(
    Math.min(
        Math.min(
            appOrderVolume - m_consumed[appOrderTypedDataHash],
            hasDataset ? datasetOrderVolume - m_consumed[datasetOrderTypedDataHash] : type(uint256).max
        ),
        workerpoolOrderVolume - m_consumed[workerpoolOrderTypedDataHash]
    ),
    requestOrderVolume - m_consumed[requestOrderTypedDataHash]
);`

Reply:
Yes, `return Math.min(..all..)`

Reply:
Done [here](https://github.com/iExecBlockchainComputing/PoCo/pull/82/commits/983d6c10c7a8e7a51c7639bf2058dfdf2a44e231)

Reply:
I should have been more explicit sorry, *if* it works, does it consume _less_ gas?

Reply:
oh gonna check that

Reply:
More gas efficient this way, thanks
`
|  IexecPoco1Delegate                                   ·  matchOrders               ·     636124  ·     848126  ·     767786  ·           77  · 
`

`|  IexecPoco1Delegate                                   ·  matchOrders               ·     635933  ·     847911  ·     767584  ·           77  ·          -  │`